workbench-core/datasets/src/s3DataSetStoragePlugin.ts

```diff
@@ -254,7 +261,12 @@ export class S3DataSetStoragePlugin implements DataSetsStoragePlugin {
         "AWS":"${externalRoleArn}"
       },
       "Action": "s3:ListBucket",
-      "Resource": "${accessPointArn}"
+      "Resource": "${accessPointArn}",
+      "Condition": {
+        "StringLike": {
+          "s3:prefix": ["${dataSetPrefix}", "${dataSetPrefix}/*"]
```

Comment:
isn't it redundant to state both, doesn't `${dataSetPrefix}/*` cover the `${dataSetPrefix}`

Reply:
if you try to access the dataset with `${dataSetPrefix}` you need the first item, but if you try to access the dataset with `${dataSetPrefix}/` you only need the second. I'm fine removing the first item if we're fine requiring a trailing slash be used to access the contents of a dataset

Reply:
isn't it a prefix though, a prefix should not indicate a dataset. In the case where the root is the datasetPrefix then it would not make sense for us to access the root, but instead the datasets under the root.

Reply:
I'm not sure I follow. Why wouldnt it make sense to access the folder defining the dataset?

Reply:
because you are access the contents of that folder

Reply:
fixed